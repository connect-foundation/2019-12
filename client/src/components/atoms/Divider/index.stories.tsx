import React from 'react';
import { text } from '@storybook/addon-knobs';

import Divider from '.';

export default {
  title: 'Atoms / Divider',
};

export const index: React.FC = () => (
  <Divider borderWidth={'1px'} type={'solid'} grayScaleLevel={6} />
);
