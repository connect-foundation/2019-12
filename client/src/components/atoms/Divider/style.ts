import styled from 'styled-components';
import { palette } from 'styled-tools';

interface DividerProps {
  borderWidth: string;
  type: 'solid' | 'dotted' | string;
  grayScaleLevel: number;
}

export const Divider = styled.hr<DividerProps>`
  border-width: ${p => p.borderWidth};
  border-style: ${p => p.type};
  border-color: ${p => palette('grayscale', p.grayScaleLevel)};
`;
