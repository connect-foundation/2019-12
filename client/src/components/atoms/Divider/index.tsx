import React from 'react';
import * as S from './style';

export interface Props {
  /** 두께 */
  borderWidth?: string;
  /** Border 스타일 */
  type?: 'solid' | 'dotted';
  /** gray scale level (color) */
  grayScaleLevel?: number;
}

function Divider({
  borderWidth = '1px',
  type = 'solid',
  grayScaleLevel = 5,
}: Props): React.ReactElement {
  return (
    <S.Divider
      borderWidth={borderWidth}
      type={type}
      grayScaleLevel={grayScaleLevel}
    />
  );
}

export default Divider;
