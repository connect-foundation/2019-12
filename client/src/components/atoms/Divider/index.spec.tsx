import React from 'react';
import { mount } from 'enzyme';
import Divider from '.';

describe('Atom / Divider', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<Divider />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
