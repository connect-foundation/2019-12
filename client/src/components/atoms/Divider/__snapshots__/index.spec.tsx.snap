// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Divider [SNAPSHOT] 렌더링 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
}

<Divider
  content="Button"
  styletype="primary"
>
  <styled.hr
    borderWidth="1px"
    grayScaleLevel={6}
    type="solid"
  >
    <StyledComponent
      borderWidth="1px"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border-width: ",
              [Function],
              ";
  border-style: ",
              [Function],
              ";
  border-color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.hr",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "hr",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      grayScaleLevel={6}
      type="solid"
    >
      <hr
        className="c0"
        type="solid"
      />
    </StyledComponent>
  </styled.hr>
</Divider>
`;
